### PR TITLE
WRQ-6229: Add ContentContainerDecorator and update calculateAndScrollTo logic in useScroll/useEvent.js

### PR DESCRIPTION
**Pull request by juwonjeong**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [x] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [x] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
When the spotlight container contains Header and VirtualGridList Item and then VirtualGridList Item gets focused, the scroll position is calculated according to the VirtualGridList item.

In some cases, it would be better to calculate the scroll position according to the container position to show the Header(which will be the VirtulaGridList title) and VirtualGridList item together.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
Add `ContentContainerDecorator` to be used in the Scroller where the developers want to calculate scroll position according to the container instead of the focused element.

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
WRQ-6229

### CODE REVIEW

File: Scroller/Scroller.js [hunk 1]
```diff
@@ -32,6 +32,7 @@ import ScrollbarPlaceholder from '../useScroll/ScrollbarPlaceholder';
 import Skinnable from '../Skinnable';
 
 import {EditableShape, EditableWrapper} from './EditableWrapper';
+import {ContentContainerDecorator} from './ContentContainerDecorator';
```
Comment (0x64) [inline Scroller/Scroller.js:35]: Let's move this line up for alphabetical order of imported module names.

File: Scroller/tests/ScrollerContentContainerDecorator-specs.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import '@testing-library/jest-dom';
```
Comment (0x64) [inline Scroller/tests/ScrollerContentContainerDecorator-specs.js:1]: Let's rename this file as we changed the name of the decorator. :)

File: samples/sampler/stories/qa/Scroller.js [hunk 3]
```diff
@@ -1,12 +1,14 @@
 import {add, is} from '@enact/core/keymap';
 import Button from '@enact/sandstone/Button';
 import BodyText from '@enact/sandstone/BodyText';
+import {FixedPopupPanels, Panel} from '@enact/sandstone/FixedPopupPanels';
+import {Heading} from '@enact/sandstone/Heading';
 import ImageItem from '@enact/sandstone/ImageItem';
 import {InputField} from '@enact/sandstone/Input';
 import Item from '@enact/sandstone/Item';
-import Scroller from '@enact/sandstone/Scroller';
+import {Scroller, ContentContainerDecorator} from '@enact/sandstone/Scroller';
```
Comment (0x64) [inline samples/sampler/stories/qa/Scroller.js:9]: Let's reorder this too just like you did for FixedPopupPanels. Thank you!

File: Scroller/Scroller.js (hunk 1 above)
Comment (juwonjeong) [reply Scroller/Scroller.js:34]: It's moved

File: Scroller/tests/ScrollerContentContainerDecorator-specs.js (hunk 2 above)
Comment (juwonjeong) [reply Scroller/tests/ScrollerContentContainerDecorator-specs.js]: Oh, I changed the file name to the ContentContainerDecorator-specs.js.

File: samples/sampler/stories/qa/Scroller.js (hunk 3 above)
Comment (juwonjeong) [reply samples/sampler/stories/qa/Scroller.js]: It's fixed.

**Review by 0x64 — APPROVED** [review]

**Comment by 0x64** [comment]
Merger's note: Travis CI did not report results but the all test are passed in Travis CI. Just technical issue of Travis CI.

**Pull request merged**